docs/website/02_examples/computePi.md

```diff
@@ -4,9 +4,132 @@ SPDX-License-Identifier: CC-BY-SA-4.0
+```
+demonstrating the usage of certain attributes of the grid module from Dune[@sander2020dune].
+Firstly, we explain the function `#!cpp boundaryUnawareRefinedCircle();`.
+
+Here, a `circleCoarse.msh`, which was created using [Gmsh](https://gmsh.info/), is read using the `Dune::GmshReader` and a `Dune::ALUGrid` object is created as shown below:
+```cpp
+constexpr int gridDim = 2;  // (1)
+using Grid            = Dune::ALUGrid<gridDim, 2, Dune::simplex, Dune::conforming>;
+auto grid             = Dune::GmshReader<Grid>::read("auxiliaryFiles/circleCoarse.msh", false);
+auto gridView         = grid->leafGridView();  // (2)
+```
+It is to note that dune-grid only supports Version 2 of the Gmsh format. The `#!cpp draw(gridView);` functionality is 
+included within the Ikarus framework to quickly draw grids and verify it for any major errors.
+The function `#!cpp grid->globalRefine(1);` is invoked to refine the grid on a global level. This means that, if we have 
+a single square-shaped 4-node quadrilateral element, the `globalRefine(1)` function will bisect the element once in 
+either direction and thereby result in a grid with 2 elements in either direction. This type of refinement is done in 
+the following and the area of the circle is compared to the value of $\pi$. The area of the circle itself is obtained by 
+summing the volume (area in 2D case) of individual elements.
+```cpp
+double area = 0.0;
+for (int i = 0; i < 3; ++i) {
+  area = 0.0;
+  grid->globalRefine(1);
+  auto gridViewRefined = grid->leafGridView();
+  std::cout << "This gridview contains: ";
+  std::cout << gridViewRefined.size(0) << " elements" << std::endl;
+  //    draw(gridViewRefined);
+  for (auto &element : elements(gridViewRefined)) {
+    area += element.geometry().volume();
+  }
+  std::cout << std::setprecision(10) << "Area: " << area << " Pi: " << std::numbers::pi << std::endl;
+}
+```
+The area of individual elements are also written in a `*.vtu` file using the module `Dune::VTKWriter`. More explanations 
+on this is found in subsequent examples. In this example, we restrict ourselves to the grid refinement strategies and 
+accessing grid entities from Dune.
+
+It is also possible to check if the element has any edge at the boundary using the method `#!cpp element.hasBoundaryIntersections()`.
+If this is true, the edges intersecting with the boundaries can be extracted and the `volume` method on this `intersection` 
+object would then return the length of the edge in the 2D case. The circumference of this unit circle is then computed as shown below:
+```cpp
+double circumference = 0.0;
+for (auto &element : elements(gridView))
+  if (element.hasBoundaryIntersections())
+    for (auto &intersection : intersections(gridView, element))
+      if (intersection.boundary()) circumference += intersection.geometry().volume();
+```
+It is observed that even though is the grid is globally refined in every loop, the value of $\pi$ computed converges to 
```

Comment:
I doesn not converge at all? It is converged in the first step since the triagnular grid does nbot change its area by refinement anymore? Thus the global refinement is not the porblem. If we would refine only on the boundary, it would still not converge since the boundary information is missing.

Reply:
Can you restated this a bit?